lib/functions/ProofUtil.ts

```diff
@@ -35,25 +43,34 @@ function partiallyValidateJWS(jws: string): void {
   }
 }
 
-function setJWSDefaults(args: JWTSignerArgs, issuerUrl: string, clientId?: string): JWTSignerArgs {
+function setJWSDefaults(args: ProofOfPossessionCallbackArgs, endpointMetadata: EndpointMetadata): { header: JWTHeader; payload: JWTPayload } {
   const now = +new Date();
-  const aud = args.payload.aud ? args.payload.aud : issuerUrl;
+  if (!endpointMetadata) {
+    throw new Error('No endpointMetadata provided');
+  }
+  const aud = endpointMetadata.issuer ? endpointMetadata.issuer : args.payload.aud;
   if (!aud) {
     throw new Error('No issuer url provided');
   }
-  const iss = args.payload.iss ? args.payload.iss : clientId;
+  if (!args.kid) {
+    throw new Error('No kid provided');
+  }
+  const iss = endpointMetadata.token_endpoint ? endpointMetadata.token_endpoint : args.payload.iss;
   if (!iss) {
     throw new Error('No clientId provided');
   }
+  const jti = args.payload.jti ? args.payload.jti : uuidv4();
```

Comment:
jti generation should be made optional. So if no jti is provided we should not automaitcally create one. JTI is fully optional

Reply:
changed this to be completely optional